### PR TITLE
jira/NMS-7698

### PR DESCRIPTION
**Pull request by christianpape**
Please review and merge fix for issue:
  http://issues.opennms.org/browse/NMS-7698

### CODE REVIEW

File: features/vaadin-dashlets/dashlet-alarms/src/main/java/org/opennms/features/vaadin/dashboard/dashlets/AlarmDetailsDashlet.java [hunk 1]
```diff
@@ -274,9 +291,25 @@ private boolean checkBoosted(List<OnmsAlarm> alarms) {
...
+         * due to restrictions in the criteria api it's quite hard
+         * to use distinct and orderBy together, so I apply a workaround
+         * to avoid alarmCb.distinct();
+         */
+
+        List<OnmsAlarm> onmsAlarmList = m_alarmDao.findMatching(alarmCb.toCriteria());
+
+        List<OnmsAlarm> distinctOnmsAlarmList = new LinkedList<>();
+        Set<Integer> onmsAlarmIdSet = new TreeSet<>();
+
```
Comment (mvrueden) [inline features/vaadin-dashlets/dashlet-alarms/src/main/java/org/opennms/features/vaadin/dashboard/dashlets/AlarmDetailsDashlet.java]: I'd prefer a HashMap here. `Map<Integer, OnmsAlarm> onmsAlarmMap`

File: features/vaadin-dashlets/dashlet-alarms/src/main/java/org/opennms/features/vaadin/dashboard/dashlets/AlarmDetailsDashlet.java [hunk 2]
```diff
@@ -274,9 +291,25 @@ private boolean checkBoosted(List<OnmsAlarm> alarms) {
...
+        for (OnmsAlarm onmsAlarm : onmsAlarmList) {
+            if (!onmsAlarmIdSet.contains(onmsAlarm.getId())) {
+                distinctOnmsAlarmList.add(onmsAlarm);
```
Comment (mvrueden) [inline features/vaadin-dashlets/dashlet-alarms/src/main/java/org/opennms/features/vaadin/dashboard/dashlets/AlarmDetailsDashlet.java]: and `if (!onmsAlarmMap.containsKey(onmsAlarm.getId())`

File: features/vaadin-dashlets/dashlet-alarms/src/main/java/org/opennms/features/vaadin/dashboard/dashlets/AlarmsDashlet.java [hunk 3]
```diff
@@ -216,9 +227,25 @@ public Component getComponent() {
...
+
+        List<OnmsAlarm> distinctOnmsAlarmList = new LinkedList<>();
+        Set<Integer> onmsAlarmIdSet = new TreeSet<>();
+
+        for (OnmsAlarm onmsAlarm : onmsAlarmList) {
+            if (!onmsAlarmIdSet.contains(onmsAlarm.getId())) {
+                distinctOnmsAlarmList.add(onmsAlarm);
+                onmsAlarmIdSet.add(onmsAlarm.getId());
+            }
+        }
```
Comment (mvrueden) [inline features/vaadin-dashlets/dashlet-alarms/src/main/java/org/opennms/features/vaadin/dashboard/dashlets/AlarmsDashlet.java]: This is the same code as above. How about re-using it?

File: features/vaadin-dashlets/dashlet-alarms/src/main/java/org/opennms/features/vaadin/dashboard/dashlets/AlarmDetailsDashlet.java (hunk 1 above)
Comment (christianpape) [inline features/vaadin-dashlets/dashlet-alarms/src/main/java/org/opennms/features/vaadin/dashboard/dashlets/AlarmDetailsDashlet.java]: The order has to be preserved. So this needs to be something 
like a linked list. Iterating over a map will destroy the sequence 
of the alarms.

Comment (j-white) [inline features/vaadin-dashlets/dashlet-alarms/src/main/java/org/opennms/features/vaadin/dashboard/dashlets/AlarmDetailsDashlet.java]: How about a [LinkedHashMap](http://docs.oracle.com/javase/7/docs/api/java/util/LinkedHashMap.html)?

File: features/vaadin-dashlets/dashlet-alarms/src/main/java/org/opennms/features/vaadin/dashboard/dashlets/AlarmsDashlet.java (hunk 3 above)
Comment (christianpape) [inline features/vaadin-dashlets/dashlet-alarms/src/main/java/org/opennms/features/vaadin/dashboard/dashlets/AlarmsDashlet.java]: The two dashlet have this small similar code abstract but this is intended.

File: features/vaadin-dashlets/dashlet-alarms/src/main/java/org/opennms/features/vaadin/dashboard/dashlets/AlarmDetailsDashlet.java (hunk 2 above)
Comment (christianpape) [inline features/vaadin-dashlets/dashlet-alarms/src/main/java/org/opennms/features/vaadin/dashboard/dashlets/AlarmDetailsDashlet.java]: Same as above. A Map will break the ordering of alarms.

File: features/vaadin-dashlets/dashlet-alarms/src/main/java/org/opennms/features/vaadin/dashboard/dashlets/AlarmDetailsDashlet.java (hunk 1 above)
Comment (christianpape) [inline features/vaadin-dashlets/dashlet-alarms/src/main/java/org/opennms/features/vaadin/dashboard/dashlets/AlarmDetailsDashlet.java]: OK, LinkedHashMap will work. I'll change it...

**Comment by soleger** [comment]
Please merge to the 16.0.1 branch. Thanks!

**Pull request merged**